Build fails on i386: cannot initialize a parameter of type 'const ttk::LongSimplexId *' (aka 'const long long *') with an lvalue of type 'vtkIdType *' (aka 'int *')

**Describe the bug**
```
/wrkdirs/usr/ports/science/ttk/work/ttk-1.1.0/core/vtk/ttkAlgorithm/ttkTriangulationFactory.cpp:280:55: fatal error: cannot initialize a parameter of type 'const ttk::LongSimplexId *' (aka 'const long long *') with an lvalue of type 'vtkIdType *' (aka 'int *')
        = triangulation->setStellarInputCells(nCells, connectivity, offsets);
                                                      ^~~~~~~~~~~~
/wrkdirs/usr/ports/science/ttk/work/ttk-1.1.0/core/base/triangulation/Triangulation.h:2464:58: note: passing argument to parameter 'connectivity' here
                                    const LongSimplexId *connectivity,
                                                         ^
1 error generated.
```

Version: 1.1.0
clang-14
FreeBSD 13.1

--- a/core/base/common/CMakeLists.txt
+++ b/core/base/common/CMakeLists.txt
@@ -25,3 +25,9 @@ if(TTK_REDUCE_TEMPLATE_INSTANTIATIONS)
   # in the base layer, only this module is affected
   target_compile_definitions(common PRIVATE TTK_REDUCE_TEMPLATE_INSTANTIATIONS)
 endif()
+
+if(CMAKE_SIZEOF_VOID_P LESS_EQUAL 4)
+  # provides compatibility between LongSimplexId and vtkIdType on i386
+  # c.f. VTK_USE_64BIT_IDS in $PARAVIEW_SOURCE_DIR/VTK/Common/Core/CMakeLists.txt
+  target_compile_definitions(common PUBLIC TTK_HW_IS_32BITS)
+endif()

--- a/core/base/common/DataTypes.h
+++ b/core/base/common/DataTypes.h
@@ -9,7 +9,11 @@
 
 namespace ttk {
   /// \brief Identifier type for simplices of any dimension.
+#ifdef TTK_HW_IS_32BITS // i386
+  using LongSimplexId = int;
+#else // amd64
   using LongSimplexId = long long int;
+#endif // TTK_HW_IS_32BITS
 
   /// \brief Identifier type for simplices of any dimension.
 #ifdef TTK_ENABLE_64BIT_IDS

